symphonia-codec-opus/src/silk/decoder.rs

```diff
@@ -0,0 +1,1102 @@
+                1 => decoder.decode_symbol_with_icdf(&constant::ICDF_LTP_FILTER_INDEX_1)?,
+                2 => decoder.decode_symbol_with_icdf(&constant::ICDF_LTP_FILTER_INDEX_2)?,
+                _ => return Err(Error::InvalidPeriodicityIndex.into()),
+            } as usize;
+
+            let filter = match periodicity_index {
+                0 => constant::CODEBOOK_LTP_FILTER_PERIODICITY_INDEX_0[filter_index],
+                1 => constant::CODEBOOK_LTP_FILTER_PERIODICITY_INDEX_1[filter_index],
+                2 => constant::CODEBOOK_LTP_FILTER_PERIODICITY_INDEX_2[filter_index],
+                _ => return Err(Error::InvalidPeriodicityIndex.into()),
+            };
+
+            subframe.copy_from_slice(&filter);
+        }
+
+        return Ok(ltp_coeffs);
+    }
+
+    fn decode_ltp_scaling<R: RangeDecoder>(&self, decoder: &mut R) -> Result<f32> {
+        const LTP_SCALE_Q14_0: f32 = 15565.0 / 16384.0;
+        const LTP_SCALE_Q14_1: f32 = 12288.0 / 16384.0;
+        const LTP_SCALE_Q14_2: f32 = 8192.0 / 16384.0;
+
+        let i = decoder.decode_symbol_with_icdf(&constant::ICDF_LTP_SCALING_PARAMETER)?;
+
+        let scale = match i {
+            0 => LTP_SCALE_Q14_0,
+            1 => LTP_SCALE_Q14_1,
+            2 => LTP_SCALE_Q14_2,
+            _ => return Err(Error::InvalidLTPScalingIndex.into()),
+        };
+
+        return Ok(scale);
+    }
+
+    /// Decodes Long-Term Prediction (LTP) parameters.
+    ///
+    /// This includes pitch lags, LTP coefficients, and LTP scaling.
+    ///
+    /// https://datatracker.ietf.org/doc/html/rfc6716#section-4.2.7.6
+    fn decode_ltp<R: RangeDecoder>(&mut self, decoder: &mut R, frame: &mut Frame) -> Result<()> {
+        let pitch_lags = self.decode_pitch_lags(decoder)?;
+        let ltp_coeffs = self.decode_ltp_coeffs(decoder)?;
+        let ltp_scale = self.decode_ltp_scaling(decoder)?;
+
+        if pitch_lags.len() != frame.subframes.len() || ltp_coeffs.len() != frame.subframes.len() {
+            return Err(Error::MismatchFrameSubframes.into());
+        }
+
+        frame.ltp_scale = ltp_scale;
+
+        frame.subframes.iter_mut()
+            .zip(pitch_lags.iter().zip(ltp_coeffs.iter()))
+            .for_each(|(subframe, (&pitch_lag, ltp_coeff))| {
+                subframe.pitch_lag = pitch_lag;
+                subframe.ltp_coeffs = ltp_coeff.to_vec();
+            });
+
+        return Ok(());
+    }
+
+    fn ltp_synthesis(excitation: &[f32], pitch_lag: u16, ltp_coeffs: &[i8], ltp_scale: f32) -> Vec<f32> {
+        let mut ltp_signal = vec![0.0; excitation.len()];
+
+        for i in 0..excitation.len() {
+            let mut pred = 0.0;
+            for (j, &coeff) in ltp_coeffs.iter().enumerate() {
+                if i >= pitch_lag as usize + j {
+                    pred += ltp_signal[i - pitch_lag as usize - j] * (coeff as f32 / 128.0);
+                }
+            }
+            ltp_signal[i] = excitation[i] + ltp_scale * pred;
```

Comment:
I hope you checked that no panic may occur

Reply:
This module may panic during development. I will debug, add tests, and replace direct indexing with safe access after refinement, as done in other modules (e.g., entropy).